python/src/aiconfig/eval/metrics.py

```diff
@@ -147,3 +152,128 @@ async def _get_overall_positive_sentiment(output_datum: str) -> TextOverallPosit
+                eval_llm_name=eval_llm_name,
+                field_descriptions_json=json.dumps(field_descriptions, sort_keys=True),
+            ),
+        ),
+    )
+
+
+def _make_openai_structured_llm_metric_helper(
+    eval_llm_name: str,
+    pydantic_basemodel_type: Type[T_BaseModel],
+    metric_name: str,
+    metric_description: str,
+    field_descriptions: dict[str, str],
+    openai_chat_completion_create: OpenAIChatCompletionCreate | None = None,
+) -> Result[Metric[str], str]:
+    schema = pydantic_basemodel_type.model_json_schema()
+    properties = schema["properties"]
+    required = schema["required"]
+
+    if not field_descriptions.keys() <= properties.keys():
+        return Err(
+            f"""
+                The following field_descriptions keys are not in the schema:
+                {set(field_descriptions.keys()) - set(properties.keys())}
+            """
+        )
+
+    def _with_description(key: str, value: dict[str, str]) -> dict[str, str]:
+        if key in field_descriptions:
+            return cu.dict_union_allow_replace(value, {"description": field_descriptions[key]})
+        return value
+
+    properties = {k: _with_description(k, v) for k, v in properties.items()}
+
+    required = required or list(properties.keys())
+
+    openai_eval_llm_chat_completion_create: common.CompletionTextToSerializedJSON = make_fn_completion_text_to_serialized_json(
+        eval_llm_name=eval_llm_name,
+        properties=properties,
+        required=required,
+        openai_chat_completion_create=(openai_chat_completion_create or default_openai_chat_completion_create),
+    )
+
+    return Ok(
+        make_structured_llm_metric(
+            openai_eval_llm_chat_completion_create,
+            eval_llm_name=eval_llm_name,
+            pydantic_basemodel_type=pydantic_basemodel_type,
+            metric_name=metric_name,
+            metric_description=metric_description,
+            field_descriptions=field_descriptions,
+        )
+    )
+
+
+def make_openai_structured_llm_metric(
+    eval_llm_name: str,
+    pydantic_basemodel_type: Type[T_BaseModel],
+    metric_name: str,
+    metric_description: str,
+    field_descriptions: dict[str, str] = {},
+    openai_chat_completion_create: OpenAIChatCompletionCreate | None = None,
+) -> Metric[str]:
+    res_metric = _make_openai_structured_llm_metric_helper(
+        eval_llm_name=eval_llm_name,
+        pydantic_basemodel_type=pydantic_basemodel_type,
+        metric_name=metric_name,
+        metric_description=metric_description,
+        field_descriptions=field_descriptions,
+        openai_chat_completion_create=openai_chat_completion_create,
+    )
+
+    # User interface: unwrap and raise
+    match res_metric:
+        case Ok(metric):
+            return metric
+        case Err(e):
+            raise ValueError(f"Error making metric: {e}")
+
+
+gpt3_5_text_ratings = make_openai_structured_llm_metric(
```

Comment:
It's an example, but it can also really be used directly off the shelf (like brevity or substring_match). I'm tracking this problem of library structure/clarity in the follow-ups doc: "straighten up modules / init / public/private / underscore / api"